core/src/main/java/io/questdb/cairo/vm/MemoryCMRImpl.java

```diff
@@ -71,8 +71,13 @@ public void extend(long newSize) {
     @Override
     public void of(FilesFacade ff, LPSZ name, long extendSegmentSize, long size, int memoryTag) {
         this.memoryTag = memoryTag;
-        openFile(ff, name);
-        map(ff, name, size);
+        if (size > -1) {
+            openFile(ff, name);
+            map(ff, name, size);
+        } else {
+            // This is read-only map and file length is -1. Means file does not exist
+            throw CairoException.instance(CairoException.ERRNO_FILE_DOES_NOT_EXIST).put("File not found: ").put(name);
```

Comment:
this is another minor spaghetti, `size` semantically is not guaranteed to be originating from `length()`. I understand the idea to cover all `length()` failures in a single place, but we might need a better semantics. For example `path` is already here. If method was called `ofFile()` etc.

Reply:
refactored in ed43dd5